### PR TITLE
apimachinery: remove unneeded GetObjectKind() impls

### PR DESCRIPTION
**Pull request by sttts**
Based on https://github.com/kubernetes/kubernetes/pull/48601

### CODE REVIEW

**Comment by sttts** [comment]
@deads2k only last commit.

**Comment by sttts** [comment]
@lavalamp @smarterclayton last commit approved?

**Comment by deads2k** [comment]
last commit lgtm

File: staging/src/k8s.io/apimachinery/pkg/runtime/register.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func (obj *TypeMeta) GroupVersionKind() schema.GroupVersionKind {
 	return schema.FromAPIVersionAndKind(obj.APIVersion, obj.Kind)
 }
 
-func (obj *Unknown) GetObjectKind() schema.ObjectKind { return &obj.TypeMeta }
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/runtime/register.go]: didn't expect this change

Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/runtime/register.go]: It embeds runtime.TypeMeta.

Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/runtime/register.go]: In fact, they are unrelated and only happen to be at the same location. Unknown embeds `runtime.TypeMeta`, and the whole point of the PR is to add `GetObjectKind` also to runtime.TypeMeta (in analogy to `metav1.TypeMeta`).

**Pull request merged**